pydantic/json_schema.py

```diff
@@ -0,0 +1,970 @@
+                try:
+                    generated[k] = self.generate_inner(v).copy()
+                except PydanticInvalidForJsonSchema:
+                    pass
+        json_schema: JsonSchemaValue = {'oneOf': list(generated.values())}
+
+        # This reflects the v1 behavior, but we may want to only include the discriminator based on dialect / etc.
+        if 'discriminator' in schema and isinstance(schema['discriminator'], str):
+            json_schema['discriminator'] = {
+                # TODO: Need to handle the case where the discriminator field has an alias
+                #   Note: Weird things would happen if the discriminator had a different alias for different choices
+                #   (This wouldn't make sense in OpenAPI)
+                'propertyName': schema['discriminator'],
+                'mapping': {k: v.get('$ref', v) for k, v in generated.items()},
+            }
+
+        return json_schema
+
+    def chain_schema(self, schema: core_schema.ChainSchema) -> JsonSchemaValue:
+        try:
+            # Note: If we wanted to generate a schema for the _serialization_, would want to use the _last_ step:
+            return self.generate_inner(schema['steps'][0])
+        except IndexError as e:
+            raise ValueError('Cannot generate a JsonSchema for a zero-step ChainSchema') from e
+
+    def lax_or_strict_schema(self, schema: core_schema.LaxOrStrictSchema) -> JsonSchemaValue:
+        # TODO: Need to read the default value off of model config or whatever
+        use_strict = schema.get('strict', False)  # TODO: replace this default False
+        if use_strict:
+            return self.generate_inner(schema['strict_schema'])
+        else:
+            return self.generate_inner(schema['lax_schema'])
+
+    def typed_dict_schema(self, schema: core_schema.TypedDictSchema) -> JsonSchemaValue:
+        properties: dict[str, JsonSchemaValue] = {}
+        required: list[str] = []
+        for name, field in schema['fields'].items():
+            # TODO: once more logic exists for alias handling, try to share it with _get_argument_name
+            if self.by_alias:
+                alias = field.get('validation_alias', name)
+                if isinstance(alias, str):
+                    name = alias
+                else:
+                    # TODO: What should be done in this case?
+                    #   Maybe tell users to override this method if they want custom behavior here?
+                    #       (If populate by name is false)
+                    pass
+            field_json_schema = self.generate_inner(field).copy()
+            if 'title' not in field_json_schema and self.field_title_should_be_set(field):
+                title = self.get_title_from_name(name)
+                field_json_schema['title'] = title
+            field_json_schema = self.handle_ref_overrides(field_json_schema)
+            properties[name] = field_json_schema
+            if field['required']:
```

Comment:
Thanks for pointing these out. They are real errors and I've been burned in other parts of the code by them :D. I figured out how to run pyright locally and will at least try to address any pyright-detected issues in the code of this PR.

Looks like there's a lot of (mostly-spurious?) things it is erroring about in other files, I'd be in favor of addressing those issues and possibly even integrating pyright into CI once v2 is released (if not before...)

Reply:
Great! :rocket: 

Sometimes PyRight can get too noisy and in some cases spurious, it might be too annoying to have it in CI if there are still too many false positives, but if there aren't too many problems, I think that's a great idea.